R/plotAbundance.R

```diff
@@ -187,12 +190,15 @@ setMethod("plotAbundance", signature = c("SummarizedExperiment"),
             stop("No data to plot. nrow(x) == 0L.", call. = FALSE)
         }
         .check_assay_present(assay.type, x)
-        if(!.is_non_empty_string(rank) && !is.null(rank)){
-            stop("'rank' must be an non empty single character value or NULL.",
+        if(!.is_non_empty_string(group) && !is.null(group)){
+            stop("'group' must be an non empty single character value or NULL.",
                 call. = FALSE)
         }
-        if(!is.null(rank)){
-            .check_taxonomic_rank(rank, x)
+        if(!is.null(group)){
+            if(!(group %in% colnames(rowData(x)))){
+                stop("'group' must be a column from rowData .",
+                     call. = FALSE)
+            }
         }
```

Comment:
Avoid nested ifs, instead use just

`!is.null(group) && !group %in% colnames(rowData(x))`